base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
```

Comment:
- [ ] Good job initializing the nameable variable. To access this variable, kindly add it as an attr_accessor :nameable in this decorator class.